libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/KeyStoreUtil.java

```diff
@@ -148,5 +159,90 @@ static X509ExtendedTrustManager createTrustManager(@Nullable KeyStore trustStore
             + "] and truststore [" + trustStore + "]");
     }
 
+    /**
+     * Creates a {@link X509ExtendedTrustManager} based on the provided certificates
+     *
+     * @param certificates the certificates to trust
+     * @return a trust manager that trusts the provided certificates
+     */
+    public static X509ExtendedTrustManager createTrustManager(Collection<Certificate> certificates) throws GeneralSecurityException {
+        KeyStore store = buildTrustStore(certificates);
+        return createTrustManager(store, TrustManagerFactory.getDefaultAlgorithm());
+    }
+
+    static Stream<KeyStoreEntry> stream(KeyStore keyStore,
+                                        Function<GeneralSecurityException, ? extends RuntimeException> exceptionHandler) {
+        try {
+            return Collections.list(keyStore.aliases()).stream().map(a -> new KeyStoreEntry(keyStore, a, exceptionHandler));
```

Comment:
Yes, because `Collections.list` forms a list from an `Enumeration<?>` which is the only useful way (I think) of turning an Enumeration (which is what Keystore returns) into a `Stream`

Reply:
In Java 9, you can turn an `Enumeration` into an `Iterator` and then an `Iterator` into a `Spliterator` and then a `Spliterator` into a `Stream`. But that's a lot of conversions just to avoid forming a list with a handful of elements in it.

Reply:
thanks